napari/plugins/hook_specifications.py

```diff
@@ -373,4 +387,19 @@ def napari_experimental_provide_dock_widget() -> Union[
         containing keyword arguments for
         :meth:`napari.qt.Window.add_dock_widget` (though note that the
         ``shortcut=`` keyword is not yet supported).
+        
+    Examples
+    --------
+    >>> from qtpy.QtWidgets import QWidget
+    >>> from napari_plugin_engine import napari_hook_implementation
+    >>> 
+    >>> class MyWidget(QWidget):
+    >>>     def __init__(self, napari_viewer):
+    >>>         self.viewer = napari_viewer
+    >>>         super().__init__()
+    >>>         # ... add Qt GUI elements accessing the viewer here
```

Comment:
Would be a bit better for this to have a button that does something, e.g. print the number of layers open in the viewer when the button is clicked?

Reply:
Indeed! I'll [make sure](https://github.com/haesleinhuepf/napari-minimal-plugin) that the examples work before continuing here.